### ISSUE TITLE
Calico doesn't upgrade when upgrading to Kops 1.8.1

### ISSUE THREAD

**Issue by jskeates** [kubernetes/kops#4559]
1. What `kops` version are you running? The command `kops version`, will display
 this information.

1.8.1

2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.

1.8.5

3. What cloud provider are you using?

AWS

4. What commands did you run?  What is the simplest way to reproduce this issue?

- Launch a new cluster with Kops 1.8.0 and Calico networking
- Without making changes to the cluster.spec, run `kops update cluster` with Kops 1.8.1
- Roll cluster instances with `kops rolling-update cluster` with Kops 1.8.1

5. What happened after the commands executed?

Cluster is still running `calico/node:v2.6.2` and `calico/kube-controllers:v1.0.0`

6. What did you expect to happen?

Cluster is running upgraded Calico pods `calico/node:v2.6.6` and `calico/kube-controllers:v1.0.3`

9. Anything else do we need to know?

It looks like the version in [`bootstrapchannelbuilder.go`](https://github.com/kubernetes/kops/blob/1.8.1/upup/pkg/fi/cloudup/bootstrapchannelbuilder.go#L480) hasn't been updated. It was included in the PR #4297, but as a second, separate commit to the YAML changes. Seems that this second commit was missed out of the cherry pick onto `release-1.8`.

**Comment by jskeates** [comment]
A workaround for this issue is to run `kubectl annotate ns kube-system addons.k8s.io/networking.projectcalico.org-`, which forces `nodeup` (I think?) to reinstall the addon. You then have to update your DaemonSet pods (e.g. by forcing a rolling update).

**Comment by chrislovecnm** [comment]
/cc @blakebarnett 

@blakebarnett did you run into something like this?

**Comment by blakebarnett** [comment]
Yeah I think it's wrong for 1.8.1, the manifest is 2.6.6 but this only says 2.6.2: https://github.com/kubernetes/kops/blob/release-1.8/upup/pkg/fi/cloudup/bootstrapchannelbuilder.go#L480

**Comment by mikesplain** [comment]
I think this is fixed in 1.9.0 since they're bumped to 2.6.7:

https://github.com/kubernetes/kops/blob/release-1.9/upup/pkg/fi/cloudup/bootstrapchannelbuilder.go#L480

**Comment by blakebarnett** [comment]
yes it's working fine in 1.9

**Comment by thomasjungblut** [comment]
anyone tested this with upgrading from k8s 1.8(.4) to 1.9(.3) using the respective kops versions? 
My calico is still stuck on 2.6.6 and didn't bother to update.

**Comment by mikesplain** [comment]
@thomaschaaf what version of kops are you using? Kops 1.9 changed calico to do a rolling update which should resolve your problem.

**Comment by thomasjungblut** [comment]
I created a cluster using Kops 1.8.1 then did the usual upgrade procedure with newly released Version 1.9.0. 
So we would probably need to patch the rolling upgrade in first and then it should trigger just fine.

**Comment by mikesplain** [comment]
Hmm that should auto updated. Does your calico config include this section? 

https://github.com/kubernetes/kops/blob/cccd71e67e681754759c5cf7f66850d8908613b6/upup/models/cloudup/resources/addons/networking.projectcalico.org/k8s-1.7.yaml.template#L126-L129

**Comment by thomasjungblut** [comment]
if I create a new 1.9 cluster from scratch yes, The old ones in 1.8.4 still run with: 
```
    "updateStrategy": {
      "type": "OnDelete"
    },
```

**Comment by thomasjungblut** [comment]
for anyone else who wants to upgrade, you can patch the daemonset up front. Save below as `calico-patch.json`:

```
{
  "spec": {
    "updateStrategy": {
      "type": "RollingUpdate",
      "rollingUpdate": {
        "maxUnavailable": 1
      }
    }
  }
}
```

then just execute the patch directly using the file: 
> kubectl patch ds calico-node -n kube-system --patch "$(cat calico-patch.json)"

if you need to force rolling update for some reason you can run this:
```
kubectl patch ds/calico-node -n kube-system -p \
  "{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"date\":\"`date +'%s'`\"}}}}}"
kubectl rollout status ds/calico-node -n kube-system
```

**Comment by mikesplain** [comment]
This is fixed in 1.9

/close
